### ISSUE TITLE
Enabled wireguard configuration should start after a reboot.

### ISSUE THREAD

**Issue by mzs114** [donaldzou/WGDashboard#360]
**Describe The Problem**
A new wireguard configuration that is enabled on the web UI should automatically run again if the server is restarted, esp when using `systemd` service or any other service manager.

![image](https://github.com/user-attachments/assets/e2c11e29-a7a6-4d47-b472-364952a0cb2f)

**Expected Error / Traceback**
```
N/A
```

**To Reproduce**
Setup 4.x WGD on Debian 12
Enable autostart with systemd or any other service manager for the above setup
Configure a new wireguard configruation and toggle as 'enable'
Reboot the host on which WGD is setup

**OS Information:**
 - OS: Debian 12
 - Python Version: 3.11.x

**Sample of your `.conf` file**
N/A

**Comment by donaldzou** [comment]
This could be done by adding a `wg-quick up wgx` command in the service file.

**Comment by mzs114** [comment]
ok, you mean in the service file of the wg-dashboard?  If possible share an example, and is it better to start it before or after wgd starts?

**Comment by donaldzou** [comment]
Yes that's right. For example, if you're following my example on the [`readme.md`](https://github.com/donaldzou/WGDashboard?tab=readme-ov-file#autostart-wgdashboard-on-boot--v22):

```ini
[Unit]
...

[Service]
...
ExecStart=/root/WGDashboard/src/wgd.sh start && wg-quick up wg0        <== Add it here
...

[Install]
...
```

I think starting before or after is fine. Hope this helps ;)

**Comment by mzs114** [comment]
> Yes that's right. For example, if you're following my example on the [`readme.md`](https://github.com/donaldzou/WGDashboard?tab=readme-ov-file#autostart-wgdashboard-on-boot--v22):
> 
> ```ini
> [Unit]
> ...
> 
> [Service]
> ...
> ExecStart=/root/WGDashboard/src/wgd.sh start && wg-quick up wg0        <== Add it here
> ...
> 
> [Install]
> ...
> ```
> 
> I think starting before or after is fine. Hope this helps ;)


I tried but get the below error:

> Job for wg-dashboard.service failed because the service did not take the steps required by its unit configuration.
See "systemctl status wg-dashboard.service" and "journalctl -xeu wg-dashboard.service" for details.

 I think the alternative also works for me atm `sudo systemctl enable wg-quick@wgX`

**Comment by opit7** [comment]
@mzs114 I did get the same error. But added the following to make it work:
```
[Unit]
...

[Service]
...
ExecStartPre=-wg-quick up wg0
...

[Install]
...
```

**Comment by donaldzou** [comment]
> @mzs114 I did get the same error. But added the following to make it work:
> 
> ```
> [Unit]
> ...
> 
> [Service]
> ...
> ExecStartPre=-wg-quick up wg0
> ...
> 
> [Install]
> ...
> ```

Great solution! Let me refer this on the documentation.

**Comment by mzs114** [comment]
> > @mzs114 I did get the same error. But added the following to make it work:
> > ```
> > [Unit]
> > ...
> > 
> > [Service]
> > ...
> > ExecStartPre=-wg-quick up wg0
> > ...
> > 
> > [Install]
> > ...
> > ```
> 
> Great solution! Let me refer this on the documentation.


Please keep a note about the implications of the two approaches:

In one it will take down the tunnel, so any connections ongoing will be dropped. 

In the other one where these are decoupled the WGdashboard can be down but the tunnel is up, allowing an operator to upgrade WGD without touching the tunnel.

**Comment by opit7** [comment]
As far as I know stopping or disabling the service has no impact on the process started with ExecStartPre.

I could not find this behavior in the offical docs but refering to chatgpt the following behavior is expected:
```
When a service has ExecStartPre directives defined and the service is either stopped or disabled, here is what happens:

1. Stopping the service:

- When a service is stopped using systemctl stop <service>, the ExecStartPre directive is not involved.

- The stopping process will follow the ExecStop or ExecStopPost directives, if defined, but the ExecStartPre commands are not executed when stopping the service.

- If the service was started and running (meaning ExecStartPre and ExecStart have both been executed before), stopping the service only affects the main service process. The ExecStartPre process (if it was a short preparatory task and completed successfully before the main process started) would already have finished and would not need to be "stopped" separately.

2. Disabling the service:

- When a service is disabled using systemctl disable <service>, it means the service is prevented from starting automatically on boot, but this action does not stop a currently running service.

- Disabling the service does not interact with ExecStartPre. It only alters the service's behavior regarding future auto-starts (e.g., after a reboot).

- You would need to run systemctl stop <service> separately if you want to stop a currently running service.


In summary:

- Stopping a service does not trigger ExecStartPre. The service will follow its stop routine (ExecStop, ExecStopPost).

- Disabling a service prevents it from starting automatically but does not stop a running service nor involve ExecStartPre. You would need to stop it manually if it is running.
```

I did stop and disable wg-dashboard.service and I can confirm a handshake is still possible even though the dashboard is down.

**Comment by mzs114** [comment]
Thank you, the above makes sense, however the original requirement was to start AND stop the tunnel along with the dashboard, when we configure `ExecStopPost` it will stop and hence my comment to keep a note about this.

**Comment by lockenkop** [comment]
I had a problem that i needed to start several interfaces with your previously mentioned methods. 
Most with no success.

tried to chain '''ExecStartPre=-wg-quick up wg0 && -wg-quick up wg1...'''
and with '''ExecStart=/root/WGDashboard/src/wgd.sh start && wg-quick up wg0 &&wg-quick up wg1'''

all of them failed so i wrote this little bash script that gets called in ExecStartPre
'''ExecStartPre=<path to script>'''

'''bash
#! /bin/bash

interfaces=("wg0" "wg1 "wg2 "wg3

for interface in ${interfaces[@]}; do
        echo activating $interface
        wg-quick up $interface
done

exit 0
'''
